### PR TITLE
Implement LOG_SLOW_WEB_LIST_FILES_MS to log web-list-files stats/errors

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

File: db/sql_changeset_schema [hunk 1]
```diff
@@ -36,8 +36,11 @@ ALTER TABLE user_accounts ADD CONSTRAINT email_uniq UNIQUE (email);
 --- alter table user_files add column zipfilesize bigint;
 --- update user_files set zipfilesize = length(data);
 CREATE INDEX user_account_devices_userid_idx on user_account_devices(userid);
+CREATE INDEX user_files_name_idx on user_files(name);
```
Comment (vshcherb) [inline db/sql_changeset_schema:39]: Too many indexes doesn't improve situation usually...

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java [hunk 2]
```diff
@@ -365,6 +365,7 @@ public ResponseEntity<String> listFiles(@RequestParam(required = false) String n
 		}
 		UserFilesResults res = userdataService.generateFiles(dev.userid, name, allVersions, true, type);
 		List <UserFileNoData> filesToIgnore = new ArrayList<>();
+		int cloudReads = 0, cacheWrites = 0;
```
Comment (vshcherb) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java:368]: this logs is ok for branch testing but too much to write to main log probably

File: db/sql_changeset_schema (hunk 1 above)
Comment (RZR-UA) [reply db/sql_changeset_schema:39]: It is not too many, it's just enough to speed up our WHERE/MAX/ORDER BY/GROUP BY query elements.

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java (hunk 2 above)
Comment (RZR-UA) [reply java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java:368]: Detailed logs are required to debug the web-list-files on production load.

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2745

File: java-tools/OsmAndServer/src/main/java/net/osmand/server/api/repo/PremiumUserFilesRepository.java [hunk 3]
```diff
@@ -38,9 +38,10 @@ public interface PremiumUserFilesRepository extends JpaRepository<UserFile, Long
 	
 	Iterable<UserFile> findAllByUserid(int userid);
 	
-	@Query("SELECT uf FROM UserFile uf " +
-			"WHERE uf.userid = :userid AND uf.name LIKE :folderName% AND uf.type = :type " +
-			"AND uf.updatetime = (SELECT MAX(uft.updatetime) FROM UserFile uft WHERE uft.userid = :userid AND uft.name = uf.name)")
+	@Query("SELECT uf FROM UserFile uf "
+			+ "WHERE uf.userid = :userid AND uf.name LIKE :folderName% AND uf.type = :type AND (uf.name, uf.updatetime) IN "
+			+ "(SELECT uft.name, MAX(uft.updatetime) FROM UserFile uft WHERE uft.userid = :userid GROUP BY uft.name)"
+	)
```
Comment (RZR-UA) [inline java-tools/OsmAndServer/src/main/java/net/osmand/server/api/repo/PremiumUserFilesRepository.java:44]: The most important part of this request.

**Comment by RZR-UA** [comment]
- [ ] add timing, only log slow list-files (>5s)

**Pull request merged**